### ISSUE TITLE
Display MsgBoxes in RTL layout if NPP local is RTL

### ISSUE THREAD

**Issue by Yaron10** [notepad-plus-plus/notepad-plus-plus#8699]
**STR:**
Select a RTL language (e.g. Hebrew)  in `Settings -> Preferences -> General -> Localization`.
Restart NPP.
Invoke a MessageBox (e.g. `Edit -> Column Mode...`).

**Result:**
The MessageBox layout is LTR.

**Expected:**
The MessageBox layout should be RTL.
***
`NativeLangSpeaker::messageBox` in `localization.cpp`.

Add
```
   if (_isRTL)
      msgBoxType |= MB_RTLREADING | MB_RIGHT;
```

before

```
   return ::MessageBox(hWnd, msg.c_str(), title.c_str(), msgBoxType);
```
***
Notepad++ v7.8.9   (32-bit)
Build time : Aug 11 2020 - 13:35:16
Path : C:\\---\notepad++.exe
Admin mode : ON
Local Conf mode : ON
OS Name : Windows 7 Enterprise (32-bit) 
OS Build : 7601.23964
Current ANSI codepage : 1255
Plugins : ComparePlus.dll HTMLTag_unicode.dll MenuIcons.dll PythonScript.dll _CustomizeToolbar.dll

**Comment by sasumner** [comment]
I'm not RTL savvy at all (as I've said before).
But, thinking about it, it seems to make a lot of sense (for RTL users).
Is this a mod that you have in your copy/fork of N++ code? If so, it works well?

**Comment by Yaron10** [comment]
@sasumner,

Yes, it works as expected.

NPP's most elements (main window, views, dialogs etc.) are "LTR/RTL aware". Why should the MsgBoxes be different?

**Comment by sasumner** [comment]
>Why should the MsgBoxes be different?

Surely they should not be different.

I wonder about "moving strings around" to different sections in the language xml files.
At first quick look it appears the msgboxes that are of the `NativeLangSpeaker::MessageBox` kind all pull from one section, unlike others that use `::MessageBox` ...

**Comment by Yaron10** [comment]
> At first quick look it appears the msgboxes that are of the NativeLangSpeaker::MessageBox kind all pull from one section, unlike others that use ::MessageBox ...

That's the point. Ideally all MsgBoxes should use `NativeLangSpeaker::messageBox`.
But it might require a lot of work changing all of them throughout the entire code.

As you understand.

Adding
```
   if (_isRTL)
      msgBoxType |= MB_RTLREADING | MB_RIGHT;
```
can still solve some of the MsgBoxes. :)

**Comment by sasumner** [comment]
>adding...can still solve some of the MsgBoxes

Yes.

Question:  You have coded stuff in Notepad++, apparently successfully...so why not submit some (prime example is THIS issue) as PRs?

**Comment by Yaron10** [comment]
> so why not submit some (prime example is THIS issue) as PRs?

I can hardly follow the rules of submitting an issue. :)

**Comment by sasumner** [comment]
@Yaron10 
Actually, what I didn't consider is that the necessary use of `git` for doing PRs is a bit of a hurdle for some.
Don't know if that is the case for you or not.
No problem, I'll do a PR for this one.

**Comment by Yaron10** [comment]
@sasumner,

:+1: 
Thank you.

**Comment by sasumner** [comment]
Changes from this issue released in Notepad++ version 7.9
